### PR TITLE
Add ability to get raw controller analog value

### PR DESCRIPTION
**Pull request by Addlibs**
Adds a bool parameter to [getAnalogControlValue](https://wiki.multitheftauto.com/wiki/GetAnalogControlState) and [getPedAnalogControlValue](https://wiki.multitheftauto.com/wiki/GetPedAnalogControlState) to poll for raw controller input for a given control value, ignoring scripted overrides, control toggles, etc.

This allows scripts to do things like mirroring analog input for accelerate onto `brake_reverse`. This will ignore non-analog inputs for the control, in other words, it'll return 0 for `accelerate` when pressing `W` on the keyboard unless you're actually pressing on a controller trigger (or whichever axis is bound to "accelerate").

### CODE REVIEW

File: Client/mods/deathmatch/logic/CClientPed.h [hunk 1]
```diff
@@ -573,6 +573,7 @@ class CClientPed : public CClientStreamElement, public CAntiCheatModule
     CStatsData*                              m_stats;
     CControllerState*                        m_currentControllerState;
     CControllerState*                        m_lastControllerState;
+    CControllerState                         m_rawControllerState;
```
Comment (qaisjp) [inline Client/mods/deathmatch/logic/CClientPed.h]: A future programmer might be confused about the difference between `m_currentControllerState` and `m_rawControllerState` (specifically, what `current` has that `raw` doesn't have). I would suggest you include comments here that explain this.

File: Client/mods/deathmatch/logic/CClientPad.cpp [hunk 2]
```diff
@@ -468,13 +468,13 @@ bool CClientPad::GetAnalogControlIndex(const char* szName, unsigned int& uiIndex
 }
 
 // Get the analog control state directly from a pad state.  Use for players.
-bool CClientPad::GetAnalogControlState(const char* szName, CControllerState& cs, bool bOnFoot, float& fState)
+bool CClientPad::GetAnalogControlState(const char* szName, CControllerState& cs, bool bOnFoot, float& fState, bool bIgnoreOverrides)
 {
     unsigned int uiIndex;
     if (GetAnalogControlIndex(szName, uiIndex))
     {
         // Do we have a script override?
```
Comment (qaisjp) [inline Client/mods/deathmatch/logic/CClientPad.cpp]: This comment should be updated to say something like:

```suggestion
        // If we are not ignoring overrides and have a script override
```

File: Client/mods/deathmatch/logic/CStaticFunctionDefinitions.cpp [hunk 3]
```diff
@@ -1660,17 +1660,21 @@ bool CStaticFunctionDefinitions::GetPedControlState(CClientPed& Ped, const char*
...
 {
     if (Ped.GetType() == CCLIENTPLAYER)
     {
         CControllerState cs;
         Ped.GetControllerState(cs);
         bool         bOnFoot = (!Ped.GetRealOccupiedVehicle());
         unsigned int uiIndex;
+
+        if (bRawInput)
+            cs = Ped.m_rawControllerState;
```
Comment (qaisjp) [inline Client/mods/deathmatch/logic/CStaticFunctionDefinitions.cpp]: If `bRawInput` is `true` then line 1668 is useless. I would suggest this:

```suggestion
            cs = Ped.m_rawControllerState;
        else
            Ped.GetControllerState(cs);
```

File: Client/mods/deathmatch/logic/CStaticFunctionDefinitions.cpp [hunk 4]
```diff
@@ -1660,17 +1660,21 @@ bool CStaticFunctionDefinitions::GetPedControlState(CClientPed& Ped, const char*
     return false;
 }
 
-bool CStaticFunctionDefinitions::GetPedAnalogControlState(CClientPed& Ped, const char* szControl, float& fState)
+bool CStaticFunctionDefinitions::GetPedAnalogControlState(CClientPed& Ped, const char* szControl, float& fState, bool bRawInput)
 {
     if (Ped.GetType() == CCLIENTPLAYER)
     {
         CControllerState cs;
         Ped.GetControllerState(cs);
```
Comment (qaisjp) [inline Client/mods/deathmatch/logic/CStaticFunctionDefinitions.cpp]:
```suggestion
```

(see above)

File: Client/mods/deathmatch/logic/CStaticFunctionDefinitions.cpp [hunk 5]
```diff
@@ -7005,13 +7009,17 @@ bool CStaticFunctionDefinitions::GetControlState(const char* szControl, bool& bS
...
+bool CStaticFunctionDefinitions::GetAnalogControlState(const char* szControl, float& fState, bool bRawInput)
 {
     CControllerState cs;
     CClientPlayer*   pLocalPlayer = m_pPlayerManager->GetLocalPlayer();
     pLocalPlayer->GetControllerState(cs);
     bool bOnFoot = (!pLocalPlayer->GetRealOccupiedVehicle());
-    if (CClientPad::GetAnalogControlState(szControl, cs, bOnFoot, fState))
+
+    if (bRawInput)
+        cs = pLocalPlayer->m_rawControllerState;
```
Comment (qaisjp) [inline Client/mods/deathmatch/logic/CStaticFunctionDefinitions.cpp]:
```suggestion
        cs = pLocalPlayer->m_rawControllerState;
    else
        pLocalPlayer->GetControllerState(cs);
```

File: Client/mods/deathmatch/logic/CStaticFunctionDefinitions.cpp [hunk 6]
```diff
@@ -7005,13 +7009,17 @@ bool CStaticFunctionDefinitions::GetControlState(const char* szControl, bool& bS
     return false;
 }
 
-bool CStaticFunctionDefinitions::GetAnalogControlState(const char* szControl, float& fState)
+bool CStaticFunctionDefinitions::GetAnalogControlState(const char* szControl, float& fState, bool bRawInput)
 {
     CControllerState cs;
     CClientPlayer*   pLocalPlayer = m_pPlayerManager->GetLocalPlayer();
     pLocalPlayer->GetControllerState(cs);
```
Comment (qaisjp) [inline Client/mods/deathmatch/logic/CStaticFunctionDefinitions.cpp]:
```suggestion
```

**Review by qaisjp — APPROVED** [review]
Thanks for the pull request! Your proposal makes sense. I've left some comments.

**Review by qaisjp — CHANGES_REQUESTED** [review]
Whoops, meant to _request changes_.

**Comment by LosFaul** [comment]
just to keep things on track
this might be connected to this stuff here:
https://github.com/multitheftauto/mtasa-blue/issues/567

**Review by qaisjp — APPROVED** [review]
lgtm, thank you! :shipit:

**Pull request merged**